File: online-javascript-angular/src/app/puzzle/Piece.ts
export class Piece {
  private index: number;
  private dataImg: string;

  constructor(index: number, dataImg: string) {
    this.index = index;
    this.dataImg = dataImg;
  }

  public getIndex(): number {
    return this.index;
  }

  public getDataImg(): string {
    return this.dataImg;
  }
}

File: offline-java/src/puzzlePxgSimulator/puzzle/Piece.java
package puzzlePxgSimulator.puzzle;

import javax.swing.Icon;

public class Piece {

	private Icon icon;
	private int index;
	
	public Piece(Icon icon, int index) {
		this.icon = icon;
		this.index = index;
	}
	
	public Icon getIcon() {
		return this.icon;
	}
	
	public int getIndex() {
		return this.index;
		
	}
}
